sdk/monitor/monitor-opentelemetry-exporter/src/export/statsbeat/longIntervalStatsbeatMetrics.ts

```diff
@@ -0,0 +1,302 @@
+  MeterProvider,
+  PeriodicExportingMetricReader,
+  PeriodicExportingMetricReaderOptions,
+} from "@opentelemetry/sdk-metrics";
+import { AzureMonitorExporterOptions, AzureMonitorStatsbeatExporter } from "../../index";
+import * as ai from "../../utils/constants/applicationinsights";
+import {
+  StatsbeatCounter,
+  StatsbeatResourceProvider,
+  STATSBEAT_LANGUAGE,
+  AIMS_URI,
+  AIMS_API_VERSION,
+  AIMS_FORMAT,
+  EU_CONNECTION_STRING,
+  EU_ENDPOINTS,
+  NON_EU_CONNECTION_STRING,
+  CommonStatsbeatProperties,
+  IVirtualMachineInfo,
+  AttachStatsbeatProperties,
+  StatsbeatFeatureType,
+  StatsbeatOptions,
+} from "./types";
+
+const os = require("os");
+
+/**
+ * Presents a singleton pattern by removing public access to the PrivateLongIntervalStatsbeatMetrics constructor.
+ */
+export class LongIntervalStatsbeatMetrics {
+  private static _instance: PrivateLongIntervalStatsbeatMetrics;
+  constructor(options: StatsbeatOptions) {
+    if (!LongIntervalStatsbeatMetrics._instance) {
+      LongIntervalStatsbeatMetrics._instance = new PrivateLongIntervalStatsbeatMetrics(options);
+    }
+  }
+
+  public getInstance() {
+    return LongIntervalStatsbeatMetrics._instance;
+  }
+}
+
+class PrivateLongIntervalStatsbeatMetrics {
+  private _statsCollectionLongInterval: number = 8600; // 1 day
+  private _isInitialized: boolean = false;
+
+  // Custom dimensions
+  private _vmInfo: IVirtualMachineInfo = {};
+  private _resourceIdentifier: string = "";
+  private _resourceProvider: string = StatsbeatResourceProvider.unknown;
+  private _os: string = os.type();
+  private _cikey: string;
+  private _runtimeVersion: string;
+  private _language: string;
+  private _version: string;
+  private _attach: string = "sdk";
+
+  private _commonProperties: CommonStatsbeatProperties;
+  private _attachProperties: AttachStatsbeatProperties;
+
+  private _feature: number = 0;
+  private _instrumentation: number = 0;
+
+  private _features: number[] = [];
+  private _instrumentations: number[] = [];
+
+  private _longIntervalStatsbeatMeterProvider: MeterProvider;
+  private _longIntervalAzureExporter: AzureMonitorStatsbeatExporter;
+  private _longIntervalMetricReader: PeriodicExportingMetricReader;
+  private _longIntervalStatsbeatMeter: Meter;
+
+  // Network Attributes
+  private _connectionString: string;
+
+  // Observable Gauges
+  private _featureStatsbeatGauge: ObservableGauge;
+  private _attachStatsbeatGauge: ObservableGauge;
+
+  constructor(options: StatsbeatOptions) {
+    this._connectionString = this._getConnectionString(options.endpointUrl);
+    const exporterConfig: AzureMonitorExporterOptions = {
+      connectionString: this._connectionString,
+    };
+
+    if (process.env.STATSBEAT_FEATURES) {
+      this._features = process.env.STATSBEAT_FEATURES.split(",").map((feature) => Number(feature));
+    }
+    if (process.env.STATSBEAT_INSTRUMENTATIONS) {
+      this._instrumentations = process.env.STATSBEAT_INSTRUMENTATIONS.split(",").map(
```

Comment:
Add a constant at the top in this file for env variables, call the env variable something like AZURE_MONITOR_STATSBEAT_FEATURES

Reply:
Same for other one, can you add a single env variable and use a JSON string?

Reply:
You can also add them in types.ts not sure if we are following that approach in this project though

Reply:
Wasn't able to access the env variables from within the `types.ts` file. Declaring the constant outside of the `LongIntervalStatsbeatMetrics` class appears to not capture any value either. Assuming this has to do with when the values in the singleton are being populated? Let me know if the updated placement works.